modules/gapi/include/opencv2/gapi/streaming/onevpl/onevpl_cfg_params.hpp

```diff
@@ -0,0 +1,60 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#ifndef OPENCV_GAPI_STREAMING_ONEVPL_ONEVPL_CFG_PARAMS_HPP
+#define OPENCV_GAPI_STREAMING_ONEVPL_ONEVPL_CFG_PARAMS_HPP
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include <opencv2/gapi/streaming/source.hpp>
+#include <opencv2/gapi/util/variant.hpp>
+
+namespace cv {
+namespace gapi {
+namespace wip {
+
+struct GAPI_EXPORTS oneVPL_cfg_param {
+    using name_t = std::string;
+    using value_t = cv::util::variant<uint8_t, int8_t,
+                                      uint16_t, int16_t,
+                                      uint32_t, int32_t,
+                                      uint64_t, int64_t,
+                                      float_t,
+                                      double_t,
+                                      void*,
+                                      std::string>;
+    template<typename ValueType>
+    static oneVPL_cfg_param create(const std::string& name, ValueType&& value, bool is_major = true) {
+        oneVPL_cfg_param param(name, oneVPL_cfg_param::value_t(std::forward<ValueType>(value)), is_major);
+        return param;
+    }
+
+    struct Priv;
+
+    const name_t& get_name() const;
+    const value_t& get_value() const;
+    bool is_major() const;
+    bool operator==(const oneVPL_cfg_param& src) const;
+    bool operator< (const oneVPL_cfg_param& src) const;
+    bool operator!=(const oneVPL_cfg_param& src) const;
```

Comment:
btw y do we need this comparison defined for operators?

also: I doubt if operator argument should be called `src`, in the operator semantic is it usually `rhs` (and `this` is `lhs`).

Reply:
it requires for sort/set_intersection/etc operations which is used for finding preferable VPL implementation by "score" (number of best matches)

ok, i will rename to rhs